discard_test.go

```diff
@@ -48,11 +48,11 @@ func TestComparable(t *testing.T) {
 		t.Fatal("any two discardLogSink must be equal")
 	}
 
-	if b := Discard().WithContextValues(ContextKey{Key: 1, Name: "one"}); b != a {
+	if b := a.WithContextValues(func(ctx context.Context) KeysAndValues { return nil }); b != a {
 		t.Fatalf("Discard logger should not get modified by WithContextValues, but it was: %+v", b)
 	}
 
-	if b := Discard().WithContext(context.Background()); b != a {
+	if b := a.WithContext(context.Background()); b != a {
```

Comment:
the change from `Discard()` to `a` should go in the previous discard commit.

Reply:
Agreed. My expectation was that I'll squash those commits once we know what we want, then it won't matter.